File: test-suite/handwritten-src/js/NestedCollectionTest.js
class NestedCollectionTest {
    constructor(module) {
        this.m = module;
    }
    setUp() {
        var jsSet1 = new Set();
        jsSet1.add("String1");
        jsSet1.add("String2");
        var jsSet2 = new Set();
        jsSet2.add("StringA");
        jsSet2.add("StringB");
        var jsList = [];
        jsList.push(jsSet1);
        jsList.push(jsSet2);
        this.jsNestedCollection = { setList: jsList };
    }
    testCppNestedRecordToJsNestedCollection() {
        var converted = this.m.testsuite.TestHelpers.getNestedCollection();
        assertEq(this.jsNestedCollection.setList, converted.setList);
    }

    testJsNestedRecordToCppNestedCollection() {
        assertTrue(this.m.testsuite.TestHelpers.checkNestedCollection(this.jsNestedCollection));
    }
}

allTests.push(NestedCollectionTest);

File: test-suite/handwritten-src/swift/NestedCollectionTest.swift
import XCTest
@testable import TestSuite
@testable import TestSuiteBridge

final class NestedCollectionTest: XCTestCase {
    let swiftNestedCollection = NestedCollection(setList: [Set(["String1", "String2"]), Set(["StringA", "StringB"])])
    
    func testCppNestedRecordToSwiftNestedCollection() throws {
        let converted = try TestHelpers_statics.getNestedCollection()
        XCTAssertEqual(swiftNestedCollection.setList, converted.setList)
    }

    func testSwiftNestedRecordToCppNestedCollection() throws {
        XCTAssertTrue(try TestHelpers_statics.checkNestedCollection(swiftNestedCollection))
    }
}
